kdl_parser/include/kdl_parser/kdl_parser.hpp

```diff
@@ -77,8 +79,10 @@ bool treeFromString(const std::string & xml, KDL::Tree & tree);
  * \param[out] tree The resulting KDL Tree
  * \return true on success, false on failure
  */
+#ifdef HAS_URDF
```

Comment:
I would rather keep the signature, but return `False` if the library was not built with urdf. That way a downstream package including `kdl_parser.hpp` doesn't have to define `HAS_URDF` themselves.

@clalancette thoughts?